.github/workflows/deploy.yml

```diff
@@ -12,98 +12,50 @@ on:
-          restore-keys: |
-            ${{ runner.os }}-buildx-
+          # This is the official way to setup secret values that are multi-line, such as an SSH private key.
+          echo 'vm_ssh_key<<EOF' >> $GITHUB_ENV
+          echo "${{ secrets.VM_SSH_KEY }}" >> $GITHUB_ENV
+          echo 'EOF' >> $GITHUB_ENV
 
-      - name: Buildx patch
-        run: |
-          Dockfiles="$(find $1  -name 'Dockerfile')"
-          d=$(date +%s)
-          i=0
-          for file in $Dockfiles; do
-            i=$(( i + 1 ))
-            echo "patching timestamp for $file"
-            touch -d @$(( d + i )) "$file"
-          done
-
-      - name: Install Cloud SDK
-        uses: google-github-actions/setup-gcloud@v0
-        with:
-          version: 367.0.0
-          service_account_key: ${{ secrets.GKE_SA_KEY }}
-          project_id: ${{ secrets.GKE_PROJECT }}
-      
-      - name: Configure Docker
-        run: gcloud auth configure-docker $CONTAINER_REGISTRY
+          echo "::set-output name=sha_short::$(git rev-parse --short HEAD)"
 
-      - name: Build and push api
-        uses: docker/build-push-action@v2
-        with:
-          context: ./
-          file: ./cmd/api/Dockerfile
-          builder: ${{ steps.buildx.outputs.name }}
-          push: true
-          tags: ${{ env.CONTAINER_REGISTRY }}/${{ secrets.GKE_PROJECT }}/textile/tableland/api:sha-${{ steps.prep.outputs.sha_short }}
-          cache-from: type=local,src=/tmp/.buildx-cache
-          cache-to: type=local,dest=/tmp/.buildx-cache
-      - name: Build and push healthbot
-        uses: docker/build-push-action@v2
+      - name: Connect via SSH, update the repo and run
+        uses: appleboy/ssh-action@master
         with:
-          context: ./
-          file: ./cmd/healthbot/Dockerfile
-          builder: ${{ steps.buildx.outputs.name }}
-          push: true
-          tags: ${{ env.CONTAINER_REGISTRY }}/${{ secrets.GKE_PROJECT }}/textile/tableland/healthbot:sha-${{ steps.prep.outputs.sha_short }}
-          cache-from: type=local,src=/tmp/.buildx-cache
-          cache-to: type=local,dest=/tmp/.buildx-cache
+          host: ${{ steps.prep.outputs.vm_host }}
+          username: validator
+          key: ${{ env.vm_ssh_key }}
+          port: 22
+          script_stop: true
+          script: |
+            cd go-tableland
+            git fetch && git checkout -f ${{ github.sha }}
+            cd docker
+            HEALTHBOT_ENABLED=true make ${{ steps.prep.outputs.deployment }}-up
+
+            # Wait to start and double-check we're running the expected version.
+            # This helps having a green check in the GH Action be high-signal that everything is fine.
+            sleep 5;
+            for i in 1 2 3 4; do [ $(curl --insecure --silent https://127.0.0.1/version | jq .GitCommit) = "\"${{ steps.prep.outputs.sha_short}}\"" ] && break || (if [ $i = 4 ]; then exit -1; else sleep 5; fi ) done 
+            echo "All healthy!"
```

Comment:
This is the step that does the deployment, which leverages the `vm_host` and `vm_ssh_key` prepared above.

At first, I thought that we could simply `./deploy.sh` some `deploy.sh` file that we prepared on each VM, but I actually think it's better to put the script explicitly in the `script:` section, since that would allow the script to be versioned and also transparent.

The steps are pretty self describing, but some notes:
- The SSH keys of the action will connect to the `validator` user in the VM, so by default in its home folder where `go-tableland` lives.
- We `cd` there, fetch and checkout the sha that we want to deploy.
- We leverage the `steps.prep.outputs.deployment` to execute `make <env>-up` with healthbot enabled.

The code in `L56-L60` intention is twofold:
- Being paranoid about docker-compose and its caching mechanism; I've seen weird things happening in the past.
- I wanted that, whenever we see a "green check" in this GH action, we are 100% sure the validator *is* running and the code we expected.

For that I leverage the new `/version` REST endpoint that I created too. We `curl` there, get the `GitCommit`, and compare it with the SHA that we theoretically deployed. These values **must** match. If that's the case, we allow the action to finish with a green check. [I added some waiting time logic just to give it some time to spin up].